### ISSUE TITLE
Auto delete of PUSH notification

### ISSUE THREAD

**Issue by arielleandrews** [bcgov/foi-flow#1514]
* As a user
* I want notifications for deleted comments to also delete
* so that I don't have notifications for comments that no longer exist 

**Assumptions & Scope**	
* This builds on previous user story #906
* Auto deletion applies to all PUSH notifications, no difference between different events that trigger the PUSH notification (parent comment, reply, reply to reply, tagged)

What is IN scope?
Auto deletion of the PUSH notification when the originating comment is deleted

What is NOT in scope?

**Acceptance Criteria**	

**Scenario 1: Auto delete notification when comment is deleted**
* GIVEN a user has posted a comment on a request that triggered a PUSH notification
* WHEN that user deletes the comment they authored
* THEN the notification sent to any other user will automatically be deleted
* AND the notification will no longer appear in the notifications log 

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**
#906 needs to be completed prior to this user story

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by m-prodan** [comment]
Tested AC as both Ministry user, and IAO user (roles reversed in both cases)

No issues, works beautifully - nice job!
